urllib3/response.py

```diff
@@ -306,7 +306,8 @@ def _error_catcher(self):
 
         with self._error_catcher():
             if amt is None:
-                data += b''.join(self.stream(decode_content))
+                async for chunk in self.stream(decode_content):
+                    data += chunk
```

Comment:
This is [accidentally quadratic](https://accidentallyquadratic.tumblr.com/). Better to instead do something like:
```python
chunks = []
async for chunk in self.stream(decode_content):
    chunks.append(chunk)
data = b''.join(chunks)
```

Reply:
Of course, sorry. Fixed.